python/cudf/cudf/core/column_accessor.py

```diff
@@ -341,6 +341,45 @@ def _pad_key(self, key: Any, pad_value="") -> Any:
             key = (key,)
         return key + (pad_value,) * (self.nlevels - len(key))
 
+    def replace_level_values(self, mapping: dict, level: int or None):
+        """
+        Returns a new ColumnAccessor with values in the keys replaced according
+        to the given mapping and level.
+        """
+        col_names, columns = list(self.keys()), list(self.values())
+
+        if self.multiindex:
+            cols_list = [list(col) for col in col_names]
+            new_names = []
+            for col_name in cols_list:
+                col_name[level] = mapping.get(col_name[level], col_name[level])
+                new_names.append(tuple(col_name))
+            ca = ColumnAccessor(
+                dict(zip(new_names, columns)),
+                level_names=self.level_names,
+                multiindex=self.multiindex,
+            )
+            self = ca
```

Comment:
Let's be cautious not to asisgn to `self` as this may throw the type checker off. We should be able to comfortably remove this line here.
```suggestion
            ca = ColumnAccessor(
                dict(zip(new_names, columns)),
                level_names=self.level_names,
                multiindex=self.multiindex,
            )
```